File: 0055-jump-game/0055-jump-game.java


class Solution {

    
    public boolean canJump(int[] nums) {
        int lastIndex = nums.length-1;
        
        for(int i = nums.length-2; i>= 0; i --){
            if(i + nums[i] >= lastIndex){
                lastIndex = i;
            }
            
            }
            
        
        
        if(lastIndex == 0){
            return true;
        }
        
        return false;
        
    }
    
}

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxjump = 0
        for i in range(len(nums)):
            if(i > maxjump):
                return False
            maxjump = max(maxjump, i + nums[i])
            if(maxjump >= len(nums)-1):
                return True
        
        return False